documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -1974,3 +1974,139 @@ The order in which the selected methods are applied is roughly determined by the
+
+    public double handicap() {
+        return handicap;
+    }
+
+    public void withHandicap(double handicap) {
+        this.handicap = handicap;
+    }
+
+    public String name() {
+        return name;
+    }
+
+    public void withName(String name) {
+        this.name = name;
+    }
+----
+====
+
+We want `GolfPlayer` to be mapped to a target object `GolfPlayerDto` similar like we 'always' do this:
+
+.Source object with fluent API.
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+@Mapper
+public interface GolfPlayerMapper {
+
+    GolfPlayerMapper INSTANCE = Mappers.getMapper( GolfPlayerMapper.class );
+
+    GolfPlayerDto golfPlayerToDto(GolfPlayer player);
+}
+----
+====
+
+This can be achieved with implementing a ServiceProviderInterface (SPI). Here's an implemented `org.mapstruct.ap.spi.AccessorNamingStrategy`:
+
+.CustomAccessorNamingStrategy
+====
+[source, java, linenums]
+[subs="verbatim,attributes"]
+----
+public class CustomAccessorNamingStrategy implements AccessorNamingStrategy {
+
+    @Override
+    public MethodType getMethodType(ExecutableElement method) {
+        if ( isGetterMethod( method ) ) {
+            return MethodType.GETTER;
+        }
+        else if ( isSetterMethod( method ) ) {
+            return MethodType.SETTER;
+        }
+        else if ( isAdderMethod( method ) ) {
+            return MethodType.ADDER;
+        }
+        else {
+            return MethodType.OTHER;
+        }
+    }
+
+    private boolean isGetterMethod(ExecutableElement method) {
+        return method.getReturnType().getKind() != TypeKind.VOID;
+    }
+
+    public boolean isSetterMethod(ExecutableElement method) {
+        String methodName = method.getSimpleName().toString();
+
+        return methodName.startsWith( "with" ) && methodName.length() > 4;
+    }
+
+    public boolean isAdderMethod(ExecutableElement method) {
+        String methodName = method.getSimpleName().toString();
+        return methodName.startsWith( "add" ) && methodName.length() > 3;
+    }
+
+    @Override
+    public String getPropertyName(ExecutableElement getterOrSetterMethod) {
+        String methodName = getterOrSetterMethod.getSimpleName().toString();
+        return Introspector.decapitalize( methodName.startsWith( "with" ) ? methodName.substring(  4 ) : methodName );
+    }
+
+    @Override
+    public String getElementName(ExecutableElement adderMethod) {
+        String methodName = adderMethod.getSimpleName().toString();
+        return Introspector.decapitalize( methodName.substring( 3 ) );
+    }
+
+    @Override
+    public String getCollectionGetterName(String property) {
+        return property.substring( 0, 1 ).toUpperCase() + property.substring( 1 );
+    }
+}
+----
+====
+Adding an SPI is as simple as adding a file named `org.mapstruct.ap.spi.AccessorNamingStrategy` with the simple content `org.mapstruct.example.CustomAccessorNamingStrategy` (FQN) to the `META-INF/services/` folder.
```

Comment:
Hmm, not _that_ simple, actually, right? Because that part needs to be in a separate project and added to the same classpath that the mapstruct-processor-*.jar is added to (either compile-classpath or processor classpath).